### PR TITLE
[Iulia Corici] functors_task_1

### PR DESCRIPTION
**Pull request by icorici**

### CODE REVIEW

**Comment by smay1613** [comment]
Please rename pull request to match the rules

File: functors_task_1/student/include/impl.h [hunk 1]
```diff
@@ -8,6 +8,17 @@
...
 {
+    BeerBrand operator()() {
+        if (currentIndex > toIndex) {
+            currentIndex = fromIndex;
+        }
+        return static_cast<BeerBrand>(currentIndex++);
+    }
+
+private:
+    int currentIndex = static_cast<int>(BeerBrand::HoeGaarden);
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: looks like should be initialized with fromIndex

File: functors_task_1/student/include/impl.h [hunk 2]
```diff
@@ -8,6 +8,17 @@
...
+    int fromIndex = static_cast<int>(BeerBrand::HoeGaarden);
+    int toIndex = static_cast<int>(BeerBrand::Leffe);
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: Please stick to the "MAX"

File: functors_task_1/student/include/impl.h [hunk 3]
```diff
@@ -16,16 +27,18 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(const BeerBrand &beerBrand)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: Pass by value
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const

File: functors_task_1/student/include/impl.h [hunk 4]
```diff
@@ -40,9 +53,20 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(const AlcoholDrink &alcoholDrink, const NonAlcoholDrink &nonAlcoholDrink)
     {
+        if (alcoholDrink == AlcoholDrink::Gin) {
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: _That's a question of style, but consider:_
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice

File: functors_task_1/student/include/impl.h (hunk 2 above)
Comment (icorici) [reply functors_task_1/student/include/impl.h]: Done

File: functors_task_1/student/include/impl.h (hunk 3 above)
Comment (icorici) [reply functors_task_1/student/include/impl.h]: Thank you!

**Pull request closed without merging**